generate_workspace/src/test/java/com/google/devtools/build/workspace/maven/VersionResolverTest.java

```diff
@@ -74,33 +73,44 @@ public void failsOnInvalidVersionRange() {
    * Asserts that given a soft pinned version specification, it selects that version,
    * and does not get the highest version. "3.4" is an example of a soft pinned version specification.
    */
-  //TODO(petros): implicitly using Maven central for resolving version. This is sketchy.
   @Test
   public void softPinnedVersions() {
```

Comment:
Here are the changed unit tests. Tried rebasing on your last pull request and that wiped out history. 

I simply assert that given a list of versions from aether, we select the highest version (i.e last element in list). I included a test for soft pins, where aether simply returns a single version.